scripts/git-run.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env sh
+set -eu
+
+source $(CDPATH= cd -- "$(dirname -- ${0})" && pwd)/common.sh
+
+assert_required_configuration_or_fail
+
+phase "Setting output workspace as safe directory ('${WORKSPACES_SOURCE_PATH}')"
+git config --global --add safe.directory "${WORKSPACES_SOURCE_PATH}"
+
+if [[ ! -z "${PARAMS_GIT_SCRIPT}" ]];
+then
+    phase "Running the provided scripts ${PARAMS_GIT_SCRIPT} in ${checkout_dir}"
+    eval "${PARAMS_GIT_SCRIPT}"
+
+    RESULT_SHA="$(git rev-parse HEAD | tr -d '\n')"
+    EXIT_CODE="$?"
+    if [ "$EXIT_CODE" != 0 ]
+    then
+        exit $EXIT_CODE
+    fi
+    # Make sure we don't add a trailing newline to the result!
+    printf "%s" "$RESULT_SHA" > "${RESULTS_COMMIT_PATH}"
+    echo $RESULT_SHA
+fi
+
+if [[ ! -z "${PARAMS_URL}" ]];
+then
+    phase "Cloning '${PARAMS_URL}' into '${checkout_dir}'"
+    set -x
+    exec git-init \
+        -url="${PARAMS_URL}" \
+        -revision="${PARAMS_REVISION}" \
+        -refspec="${PARAMS_REFSPEC}" \
+        -path="${checkout_dir}" \
+        -sslVerify="${PARAMS_SSL_VERIFY}" \
+        -submodules="${PARAMS_SUBMODULES}" \
+        -depth="${PARAMS_DEPTH}" \
+        -sparseCheckoutDirectories="${PARAMS_SPARSE_CHECKOUT_DIRECTORIES}"
+fi
```

Comment:
Given the script either run `git clone` (`git-init`) or an arbitrary "git script", this block should become a `if/else`.